src/simsopt/geo/surface.py

```diff
@@ -839,6 +839,130 @@ def interpolate_on_arclength_grid(self, function, theta_evaluate):
 
         return function_interpolated
 
+    def fourier_transform_field(self, field, mpol=None, ntor=None, normalization=None, **kwargs):
```

Comment:
one suggestion to simplify this function is to use a prewritten FFT and inverse FFT algorithm here